### ISSUE TITLE
Emulator Pause Issue

### ISSUE THREAD

**Issue by ReyVGM** [PCSX2/pcsx2#1734]
I think it's not necessary for me to provide any specs since this is probably something that happens regardless.

Anyway. If you are playing on full screen  (Alt+Enter) and use the emulator pause option (I have it set up  to a keyboard key), then instead of pausing in full screen, the emulator will go to windowed mode and stay like that until you resume.

I'm using D3D11 SW, all other options are default.

**Comment by FlatOutPS2** [comment]
What happens if you go to Config > Emulation Settinfgs > GS Window tab, enable the "Default to fullscreen on open" option, and then pause it with the shortcut?

**Comment by avih** [comment]
This is by (my) design I think, though TBH I don't recall the reasons now. IIRC something relating to running without GUI and confirmation. But I don't quite recall. Hopefully I wrote some useful commit messages to explain the reasons.

**Comment by avih** [comment]
OK, I just checked, assuming you set the video window to not hide on pause, it indeed gets to windowed mode on pause and then goes to fullscreen again when you resume.

I think it's reasonable behavior TBH, both because I'm not fond of leaving a paused emulation window at fullscreen without any visible indication that it's paused or instructions how to resume or exit it, and also when you run without GUI, it's the only way to exit emulation by closing the window with the "X" button.

I still don't recall the all the considerations now, but I also still think it's reasonable behavior.

Shall we close it as WONTFIX?

**Comment by ReyVGM** [comment]
Would it be possible to include as an option to pause the emulator in full screen without it getting windowed?

Enabling "Default to fullscreen on open" as FlatOutPS2 suggested still has the behavior. I understand your reasons Avih, but having the option to override that behavior would be nice too for people that use the pause/resume properly.

**Comment by avih** [comment]
It's always an option, just one I prefer not to implement for the reasons I've mentioned, and because it's sensitive and can easily break some use cases. If you really must pause and stay at full screen, just pause in-game.

**Comment by ReyVGM** [comment]
You can't pause games during cutscenes or other instances. Emulator pause is useful when you are taking notes or trying to analyze a scene.

**Comment by avih** [comment]
> You can't pause games during cutscenes or other instances. Emulator pause is useful when you are taking notes or trying to analyze a scene.

Which you can do perfectly fine also when it's not fullscreen. The best I can offer is to maximize before going fullscreen, such that on pause it will be maximized rather than smaller window. It's not exactly what you wanted, but I'm sure you could live with that.

**Comment by FlatOutPS2** [comment]
You can open one the plugin settings windows or try to swith between ISO/Plugin/No Disc. That will pause the emulation without leaving full screen, and you can just click cancel to continue.

**Comment by ReyVGM** [comment]
Yes, the pause works fine on windowed mode. That's not a problem. 

The solution you are offering is not really ideal, because you are still taken out of full screen mode. I'm not really looking to have anyone spend time implementing something if it's not ideal. Either it's implemented properly (like every other emulator out there), or it's best not to add it.

FlatOutPS2:
I have the Pause/resume option set up to a keyboard key for maximum quickness and convenience, so such option is not really ideal.

Anyway, thanks for all the suggestions. But if avih doesn't feel comfortable allowing pause while staying in full screen, then there's really nothing else to talk about.

**Comment by FlatOutPS2** [comment]
> I think it's reasonable behavior TBH, both because I'm not fond of leaving a paused emulation window at fullscreen without any visible indication that it's paused or instructions how to resume or exit it, and also when you run without GUI, I think it's the only way to exit emulation by closing the window with the "X" button.

Keeping full screen only when running with GUI, and printing a pause message on the OSD once that PR has been merged would avoid those issues.

**Comment by avih** [comment]
> and printing a pause message on the OSD once that PR has been merged would avoid those issues.

The thought of using the OSD, once it's merged, did cross my mind. But like I said, IIRC the logic in this area is delicate. Once the OSD land, if there's a PR with simple enough implementation, we could reconsider.

**Comment by FlatOutPS2** [comment]
The OSD wouldn't really work for this currently, so the following solution is probably the easiest way to possibly implement this(disable "Hide window when paused", enter fullscreen, then pause the emulation):

[pcsx2 - pause test.zip](https://github.com/PCSX2/pcsx2/files/693748/pcsx2.-.pause.test.zip)

**Comment by RedDevilus** [comment]
Seems this is still active:
- Boot in Windowed
- Alt+Enter or other method to go fullscreen
- Pause
- Resume 
- Boots in Windowed

**Comment by NoahSchiffman** [comment]
Hi - I'm someone else that discovered this thread while searching for an answer, and it seems as though this is the closest place I can post. Every time that I press the spacebar on my emulation, it freezes. I'm using the nightly build from yesterday. Yesterday being 9/2/2022.

**Comment by refractionpcsx2** [comment]
then unmap space from pause.

**Comment by Mrlinkwii** [comment]
can this be checked on master ?

**Comment by JordanTheToaster** [comment]
Doesn't happen on v1.7.3980.
